Source/WebCore/style/AnchorPositionEvaluator.cpp

```diff
@@ -761,13 +761,20 @@ static bool firstChildPrecedesSecondChild(const RenderObject* firstChild, const
 // returns nullptr.
 // (*): an anchor element can also establish an anchor scope containing itself. In this
 // case, the return value is itself.
-static CheckedPtr<const Element> anchorScopeForAnchorElement(const Element& anchorElement, const Vector<ScopedName>& anchorNames)
+static CheckedPtr<const Element> anchorScopeForAnchorName(const RenderBoxModelObject& anchorRenderer, const AtomString anchorName)
 {
-    // Precondition: anchorElement is an anchor, which has at least one anchor name.
-    ASSERT(!anchorNames.isEmpty());
+    // Precondition: anchorElement is an anchor, which has the specified name.
+    bool namecheck = false;
+    for (const auto& scopedName : anchorRenderer.style().anchorNames()) {
+        if (scopedName.name == anchorName)
+            namecheck = true;
```

Comment:
You can probably break here, but I think using contains()/containsIf() is probably nicer.